handler.go

```diff
@@ -452,20 +460,50 @@ func invoke(ctx context.Context, sf ServableFunction, input *sdkrequest.Request)
 
 	// If we have an actual value to add to the event, vs `Input[any]`, set it.
 	if sf.ZeroEvent() != nil {
+		eventType := reflect.TypeOf(sf.ZeroEvent())
+
 		// Create a new copy of the event.
-		evtPtr := reflect.New(reflect.TypeOf(sf.ZeroEvent())).Interface()
+		evtPtr := reflect.New(eventType).Interface()
 		if err := json.Unmarshal(input.Event, evtPtr); err != nil {
 			return nil, nil, fmt.Errorf("error unmarshalling event for function: %w", err)
 		}
 		evt := reflect.ValueOf(evtPtr).Elem()
 		inputVal.FieldByName("Event").Set(evt)
+
+		// events
+		sliceType := reflect.SliceOf(eventType)
+		evtList := reflect.MakeSlice(sliceType, 0, len(input.Events))
+
+		for _, rawjson := range input.Events {
+			newEvent := reflect.New(eventType).Interface()
+
+			if err := json.Unmarshal(rawjson, &newEvent); err != nil {
+				return nil, nil, fmt.Errorf("non-zero event: error unmarshalling event in event list: %w", err)
+			}
+
+			evtList = reflect.Append(evtList, reflect.ValueOf(newEvent).Elem())
+		}
+		inputVal.FieldByName("Events").Set(evtList)
 	} else {
 		// Use a raw map to hold the input.
 		val := map[string]any{}
 		if err := json.Unmarshal(input.Event, &val); err != nil {
 			return nil, nil, fmt.Errorf("error unmarshalling event for function: %w", err)
 		}
 		inputVal.FieldByName("Event").Set(reflect.ValueOf(val))
+
+		// events
+		events := make([]any, len(input.Events))
```

Comment:
I've literally never unmarshalled into `[]any` - what type does this produce in the function?  I'm thinking that for consistency this should be the same as a regular event: `[]map[string]any`?

Reply:
without making it to `[]any` the tests will fail here with a panic because the type signature doesn't match.
https://github.com/inngest/inngestgo/blob/531c44bfbee94502e24ca33e9e53d1d89d18ba3c/handler_test.go#L179-L232

Reply:
this is what it looks like if I set it to `[]map[string]any`
![reflectfailure](https://github.com/inngest/inngestgo/assets/5746693/b1a2ca99-cfba-488b-a9bd-00397c38a237)